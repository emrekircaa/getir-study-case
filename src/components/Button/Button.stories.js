import { Button } from ".";

const meta = {
  title: "Components/Button",
  component: Button,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
};

export default meta;

export const sample = {
  args: {
    children: "Fist story",
  },
};
