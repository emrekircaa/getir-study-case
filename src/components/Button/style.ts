import styled from "styled-components";
import { colors } from "../../theme";

export const StyledButton = styled.button`
  width: 100%;
  height: 22px;
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 2px;
  font-weight: 600;
  font-size: 12px;
  line-height: 20px;
  border: none;
  cursor: pointer;
  background-color: ${colors.primary};
  color: white;
  &:hover {
    opacity: 0.8;
  }
`;
