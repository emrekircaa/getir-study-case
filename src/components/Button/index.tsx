import React from "react";
import { StyledButton } from "./style";
interface ButtonProps {
  children: React.ReactNode;
  onClick: () => void;
}

export const Button: React.FC<ButtonProps> = ({ children, onClick }) => {
  return <StyledButton onClick={onClick}>{children}</StyledButton>;
};
