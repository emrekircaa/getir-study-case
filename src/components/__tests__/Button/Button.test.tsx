import { render, fireEvent } from "@testing-library/react";
import { Button } from "../../Button/index";

describe("Button component", () => {
  it("renders correctly with default props", () => {
    const handleClick = jest.fn();
    const { container } = render(
      <Button onClick={handleClick}>Default Button</Button>
    );

    // Save the rendered output to a snapshot file
    expect(container).toMatchSnapshot();
  });

  it("calls the onClick handler when clicked", () => {
    const handleClick = jest.fn();

    const { getByText } = render(
      <Button onClick={handleClick}>Custom Button</Button>
    );

    // Get the button element by its text content
    const button = getByText(/custom button/i);

    // Fire a click event on the button
    fireEvent.click(button);

    // Verify that the onClick handler was called once
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
