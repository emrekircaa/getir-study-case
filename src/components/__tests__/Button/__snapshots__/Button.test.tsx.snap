// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly with default props 1`] = `
<div>
  <button
    class="sc-beySPh hqIXqv"
  >
    Default Button
  </button>
</div>
`;
